pylint/lint.py

```diff
@@ -1054,31 +1082,57 @@ def _do_check(self, files_or_modules):
             if interfaces.implements(checker, interfaces.IAstroidChecker):
                 walker.add_checker(checker)
         # build ast and check modules or packages
-        for descr in self.expand_files(files_or_modules):
-            modname, filepath, is_arg = descr["name"], descr["path"], descr["isarg"]
-            if not self.should_analyze_file(modname, filepath, is_argument=is_arg):
-                continue
+        if self.config.from_stdin:
+            assert len(files_or_modules) == 1
```

Comment:
Maybe raise a more obvious exception? if we don't have anything, you can add one in `pylint.exceptions`.